src/hotspot/share/sanitizers/address.hpp

```diff
@@ -76,6 +76,33 @@
   } while (false)
 #endif
 
+template<typename T>
+class AsanPoisoningHelper {
+  const T* _memory_region;
+ public:
+  AsanPoisoningHelper() = delete;
+  AsanPoisoningHelper(const T* addr) : _memory_region(addr) {
+    #if INCLUDE_ASAN
```

Comment:
We use ADDRESS_SANITIZER in other places in this file, so I would use that.

In addition, I would wrap the whole class into ADDRESS_SANITIZER and not do this for the individual functions. And provide a simple stub for the non-ASAN case.